modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -0,0 +1,463 @@
+      end
+
+      alg = cert.signature_algorithm
+
+      if alg.downcase.include? 'md5'
+        print_status("\tWARNING: Signature algorithm using MD5 (#{alg})")
+      end
+
+      vhostn = nil
+      cert.subject.to_a.each do |n|
+        vhostn = n[1] if n[0] == 'CN'
+      end
+
+      if vhostn
+        print_status("\tHas common name #{vhostn}")
+
+        # Store the virtual hostname for HTTP
+        report_note(
+          host: ip,
+          port: rport,
+          proto: 'tcp',
+          type: 'http.vhost',
+          data: { name: vhostn }
+        )
+
+        # Update the server hostname if necessary
+        if vhostn !~ /localhost|snakeoil/i
+          report_host(
+            host: ip,
+            name: vhostn
+          )
+        end
+
+      end
+    else
+      print_status("\tNo certificate subject or common name found.")
+    end
+  end
+
+  def check_vulnerabilities(ip, ssl_version, ssl_cipher, cert)
+    # POODLE
+    if ssl_version == 'SSLv3'
+      print_good('Accepts SSLv3, vulnerable to POODLE')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv3 is available. Vulenrable to POODLE, CVE-2014-3566.",
+        refs: 'CVE-2014-3566',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # DROWN
+    if ssl_version == 'SSLv2'
+      print_good('Accepts SSLv2, vulnerable to DROWN')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv2 is available. Vulenrable to DROWN, CVE-2016-0800.",
+        refs: 'CVE-2016-0800',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # BEAST
+    if ((ssl_version == 'SSLv3') || (ssl_version == 'TLSv1.0')) && ssl_cipher.include?('CBC')
+      print_good('Accepts SSLv3/TLSv1 and a CBC cipher, vulnerable to BEAST')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv3/TLSv1 and a CBC cipher. Vulenrable to BEAST, CVE-2011-3389.",
+        refs: 'CVE-2011-3389',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # RC4 ciphers
+    if ssl_cipher.upcase.include?('RC4')
+      print_good('Accepts RC4 cipher.')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed RC4 cipher. CVE-2013-2566.",
+        refs: 'CVE-2013-2566',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # export ciphers
+    if ssl_cipher.upcase.include?('EXPORT')
+      print_good('Accepts EXPORT based cipher.')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed EXPORT based cipher.",
+        refs: references,
```

Comment:
This is an entire array of references??? Is this really what we want here given many of these don't even seem related?

```
references
[#<Msf::Module::SiteReference:0x0000557cf5ab0b68 @ctx_id="URL", @ctx_val="https://security.googleblog.com/2014/10/this-poodle-bites-exploiting-ssl-30.html", @site="https://security.googleblog.com/2014/10/this-poodle-bites-exploiting-ssl-30.html">, #<Msf::Module::SiteReference:0x0000557cf5ab06e0 @ctx_id="CVE", @ctx_val="2014-3566", @site="https://nvd.nist.gov/vuln/detail/CVE-2014-3566">, #<Msf::Module::SiteReference:0x0000557cf5ab05f0 @ctx_id="URL", @ctx_val="https://www.openssl.org/~bodo/ssl-poodle.pdf", @site="https://www.openssl.org/~bodo/ssl-poodle.pdf">, #<Msf::Module::SiteReference:0x0000557cf5ab0280 @ctx_id="URL", @ctx_val="https://datatracker.ietf.org/doc/rfc8996/", @site="https://datatracker.ietf.org/doc/rfc8996/">, #<Msf::Module::SiteReference:0x0000557cf5aabfa0 @ctx_id="URL", @ctx_val="https://www.win.tue.nl/hashclash/rogue-ca/", @site="https://www.win.tue.nl/hashclash/rogue-ca/">, #<Msf::Module::SiteReference:0x0000557cf5aabc08 @ctx_id="URL", @ctx_val="https://drownattack.com/", @site="https://drownattack.com/">, #<Msf::Module::SiteReference:0x0000557cf5aab870 @ctx_id="CVE", @ctx_val="2016-0800", @site="https://nvd.nist.gov/vuln/detail/CVE-2016-0800">, #<Msf::Module::SiteReference:0x0000557cf5aab6b8 @ctx_id="CVE", @ctx_val="2011-3389", @site="https://nvd.nist.gov/vuln/detail/CVE-2011-3389">, #<Msf::Module::SiteReference:0x0000557cf5aab438 @ctx_id="URL", @ctx_val="http://www.isg.rhul.ac.uk/tls/", @site="http://www.isg.rhul.ac.uk/tls/">, #<Msf::Module::SiteReference:0x0000557cf5aab118 @ctx_id="CVE", @ctx_val="2013-2566", @site="https://nvd.nist.gov/vuln/detail/CVE-2013-2566">, #<Msf::Module::SiteReference:0x0000557cf5aab050 @ctx_id="CVE", @ctx_val="2015-4000", @site="https://nvd.nist.gov/vuln/detail/CVE-2015-4000">]
```

This also applies to at least two entries lower down in this function.

Reply:
This made more sense when we were just checking the certificate itself (previous module), or just POODLE (previous module).  I agree we're throwing a lot of crud in there that isn't related.